### ISSUE TITLE
Allow users to block any tags of their choice from appearing in news feed or search

### ISSUE THREAD

**Issue by brirush84** [iftechfoundation/ifdb-suggestion-tracker#160]
This would allow users to customize their feed to, for instance, avoid games tagged 'adult'.

**Comment by qdacsvx** [comment]
This would allow anyone to block any game they like by tagging it indiscriminately.

**Comment by curiousdannii** [comment]
Once mods can remove tags they'll be able to watch for and fix bad tags.

**Comment by qdacsvx** [comment]
@curiousdannii a person might just want to disappear a game and not care if it is 100% 10% or 1% effective. I've asked @dfabulich to look at this.

Most tags are incomplete in the sense that there are some, many, or most!, works which should have the tag but do not. This means using a tag for filtering will likely result in leaks when something which isn't tagged but should be escapes the filter.

@brirush84 how would you feel if you sometimes saw 'adult' games? Would you tag them 'adult' to help out others? It would help the tag to be more complete if you did.

Even if there was a complete tag, as soon as a new work is added it could become incomplete if the tag should be added to the new work but the tag is not added.

**Comment by brirush84** [comment]
I see what you mean. I intend this mainly as a customization option so people can filter out things they don't like. I see what you mean about malicious filtering and I've seen people do stuff like that before with games (like tagging 'game doesn't exist' on a game that exists on itch but had a bad url in the download box). 

Two things could help with avoiding this: the first would be a little message ('23 results not shown due to your filters'). The second is what you said, mods making sure tags are correct by either adding/subtracting tags or by finding a way to monitor new tags and correct any improper ones.

**Comment by qdacsvx** [comment]
If works with the filtered tag were dehighlighted but not hidden it would reduce the incentive to add possibly filtered tags indiscriminately in an attempt to hide them.

Another desideratum for a leaky filter is neutrality. Are the works leaked by the filter (i.e. false negatives which should have been filtered) a fair sample or are some works more likely to be leaked than others (bias)?

A parallel question for a tag: If I click on 'horror' tag and get the search result for all works with that tag, will it be a fair sample of all the horror works on ifdb? In general, the answer is no. Most tags are biased because tagging depends on attention.

In conclusion, a filter that depends on a tag will likely leak works in a biased way.

This is important because it could create a false impression of the e.g. horror works leading someone using the filter to believe that horror works on ifdb are better/worse than they really are.

**Comment by qdacsvx** [comment]
>I see what you mean about malicious filtering and I've seen people do stuff like that before with games (like tagging 'game doesn't exist' on a game that exists on itch but had a bad url in the download box).

@brirush84 ask Peter Piers for lore about old tags and old practices on ifdb.

The title of this suggestion "Block tags... " is suggestive. It's not really about blocking tags; it's really about blocking games. When I read in a forum thread that people think that political views different to their own are "adult", it seems to me an "adult" filter is really a "I don't like it" filter.

People block things all day on social media. Most adults can identify an adult interactive fiction. Let people block things they don't like on ifdb. If blocking is easier than rating, some downvotes could be eliminated.
